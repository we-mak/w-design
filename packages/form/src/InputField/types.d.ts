import {
  GlobProps,
  SizeProps,
  SyntheticEventProps,
  KeyboardEventProps,
  InputControlProps
} from "@w-design/core";

export interface InputFieldProps
  extends GlobProps,
    SizeProps,
    SyntheticEventProps,
    KeyboardEventProps,
    InputControlProps {
  name?: string;
  type?: string;
  /* auto-complete attribute for input */
  autoComplete?: string;
  appearance?: "standard" | "none";
  isRequired?: boolean;
  isError?: boolean;
  isValid?: boolean;
  isWarning?: boolean;
  isFocus?: boolean;
  isDisabled?: boolean;
  tabIndex?: number;
  /** input ref*/
  inputRef?: (ref: HTMLInputElement) => {};
}
