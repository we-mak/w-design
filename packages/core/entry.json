{
  "index": "src/index.ts",
  "AnchorScroll": "src/AnchorScroll/index.tsx",
  "Avatar": "src/Avatar/index.tsx",
  "AvatarGroup": "src/AvatarGroup/index.tsx",
  "Box": "src/Box/index.ts",
  "Breadcrumbs": "src/Breadcrumbs/index.tsx",
  "Button": "src/Button/index.tsx",
  "Card": "src/Card/index.tsx",
  "Comment": "src/Comment/index.tsx",
  "Container": "src/Container/index.tsx",
  "CustomTag": "src/CustomTag/index.tsx",
  "DisplayElement": "src/DisplayElement/index.tsx",
  "Divider": "src/Divider/index.ts",
  "Drawer": "src/Drawer/index.tsx",
  "Dropdown": "src/Dropdown/index.tsx",
  "EmptyView": "src/EmptyView/index.tsx",
  "Flexbox": "src/Flexbox/index.tsx",
  "Icon": "src/Icon/index.tsx",
  "Image": "src/Image/index.tsx",
  "List": "src/List/index.tsx",
  "Menu": "src/Menu/index.tsx",
  "Modal": "src/Modal/index.tsx",
  "Navbar": "src/Navbar/index.tsx",
  "Picture": "src/Picture/index.tsx",
  "Portal": "src/Portal/index.tsx",
  "ProgressLoader": "src/ProgressLoader/index.tsx",
  "Provider": "src/Provider/index.tsx",
  "PushMessage": "src/PushMessage/index.tsx",
  "SectionMessage": "src/SectionMessage/index.tsx",
  "Spinner": "src/Spinner/index.tsx",
  "Typo": "src/Typo/index.tsx"
}
