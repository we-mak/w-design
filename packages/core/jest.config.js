const base = require("../../jest.config.base.js");
const pack = require("./package.json");

module.exports = {
  ...base,
  name: pack.name,
  displayName: pack.name,
  rootDir: "../..",
  testMatch: [
    `<rootDir>/packages/core/**/*.test.+(ts|tsx|js)`,
    `<rootDir>/packages/core/**/__tests__/*.+(ts|tsx|js)`
  ]
};
