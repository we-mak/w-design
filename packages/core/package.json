{
  "name": "@w-design/core",
  "version": "0.41.0-alpha.28+d990095",
  "description": "w-design core components",
  "keywords": [
    "w-design",
    "react",
    "core",
    "components"
  ],
  "author": "Son_dang <son.dang@doda100.com >",
  "homepage": "https://github.com/we-mak/w-design#readme",
  "license": "MIT",
  "main": "lib/cjs/index.js",
  "module": "lib/es/index.es.js",
  "types": "lib/cjs/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib/cjs/*",
    "lib/es/*"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://viiiprock@github.com/we-mak/w-design.git"
  },
  "scripts": {
    "test": "cross-env CI=1 jest test -u --colors"
  },
  "bugs": {
    "url": "https://github.com/we-mak/w-design/issues"
  },
  "dependencies": {
    "@types/body-scroll-lock": "^2.6.1",
    "@w-design/helpers": "^0.40.0",
    "@w-design/hooks": "^0.40.0",
    "body-scroll-lock": "^2.6.4"
  },
  "gitHead": "d990095e2f75e0d6e34aff66a98969a4de8eae6f"
}
