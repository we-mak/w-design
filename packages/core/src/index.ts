export { default as AnchorScroll } from "./AnchorScroll";
export { default as Avatar } from "./Avatar";
export { default as AvatarGroup } from "./AvatarGroup";
export { default as Badge } from "./Badge";
export { default as Box } from "./Box";
export { default as Breadcrumbs, BreadcrumbsItem } from "./Breadcrumbs";
export { default as Button, ButtonGroup } from "./Button";
export {
  default as Card,
  CardMedia,
  CardHeader,
  CardContent,
  CardFooter,
} from "./Card";

export * from "./Chip";
export { default as Chip } from "./Chip";

export { default as Collapse } from "./Collapse";
export { default as Comment } from "./Comment";
export { default as Container } from "./Container";
export { default as CustomTag } from "./CustomTag";
export { default as DisplayElement } from "./DisplayElement";
export { default as Divider } from "./Divider";
export { default as Drawer } from "./Drawer";
export { default as Dropdown } from "./Dropdown";
export { default as EmptyView } from "./EmptyView";
export { default as Flexbox } from "./Flexbox";
export { default as Icon } from "./Icon";
export { default as Image, ImageRatio } from "./Image";
export { default as List, ListItem, ListFooter } from "./List";
export { default as Menu, MenuItem, MenuHeading, SubMenu } from "./Menu";
export { default as Modal } from "./Modal";
export { default as Navbar } from "./Navbar";

export * from "./Pagination";
export { default as Pagination } from "./Pagination";

export { default as Picture } from "./Picture";
export { default as Portal } from "./Portal";

export * from "./Popover";
export { default as Popover } from "./Popover";

export { default as ProgressLoader } from "./ProgressLoader";
export { default as Provider } from "./Provider";
export { default as SectionMessage } from "./SectionMessage";

export * from "./Spinner";
export { default as Spinner } from "./Spinner";

export * from "./Steps";
export { default as Steps } from "./Steps";

export { default as Tabs, Tab } from "./Tabs";
export { default as Typo } from "./Typo";

export * from "./PushMessage";
export { default as PushMessage } from "./PushMessage";
