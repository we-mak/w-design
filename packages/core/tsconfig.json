{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib",
    "declarationDir": "lib",
    "baseUrl": "."
  },
  "include": ["src/**/*"]
}
