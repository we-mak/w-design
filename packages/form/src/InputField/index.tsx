/**
 * Input Field
 * Single input field
 */
import * as React from "react";
import styled from "styled-components";
import { getInputStyle } from "./getInputStyle";
import {
  GlobProps,
  SizeProps,
  SyntheticEventProps,
  KeyboardEventProps,
  InputControlProps
} from "../../../types/typings";

export interface InputFieldProps
  extends GlobProps,
    SizeProps,
    SyntheticEventProps,
    KeyboardEventProps,
    InputControlProps {
  name?: string;
  type?: string;
  /* auto-complete attribute for input */
  autoComplete?: string;
  appearance?: "standard" | "none";
  isRequired?: boolean;
  isError?: boolean;
  isValid?: boolean;
  isWarning?: boolean;
  isFocus?: boolean;
  isDisabled?: boolean;
  tabIndex?: number;
  /** input ref*/
  inputRef?: (ref: HTMLInputElement) => {};
}

const StyledInput = styled.input<any>`
  ${getInputStyle};
`;
StyledInput.displayName = "StyledInput";

const InputField = (props: InputFieldProps) => {
  const {
    inputRef,
    appearance,
    type = "text",
    name,
    className,
    id,
    autoComplete = "off",
    size,
    tabIndex,
    value,
    defaultValue,
    isRequired,
    isError,
    isValid,
    isWarning,
    isFocus = false,
    isDisabled = false,
    onChange,
    onBlur,
    ...rest
  } = props;

  let input!: HTMLInputElement | undefined;

  const getInputRef = (ref: HTMLInputElement) => {
    input = ref;
    if (inputRef) inputRef(ref);
  };

  React.useEffect(() => {
    if (isFocus && input) input.focus();
    return () => (input = undefined);
  });

  return (
    <StyledInput
      ref={getInputRef}
      type={type}
      size={size}
      name={name}
      appearance={appearance}
      autoComplete={autoComplete}
      className={className}
      id={id}
      required={isRequired}
      value={value}
      defaultValue={defaultValue}
      isFocus={isFocus}
      isValid={isValid}
      isError={isError}
      isWarning={isWarning}
      disabled={isDisabled}
      onChange={isDisabled ? undefined : onChange}
      onBlur={onBlur}
      tabIndex={tabIndex}
      {...rest}
    />
  );
};

export default InputField;
