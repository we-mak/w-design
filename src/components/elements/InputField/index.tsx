/**
 * Input Field
 * Single input field
 */
import * as React from "react";
import styled from "styled-components";
import { InputFieldProps } from "./types";
import { getInputStyle } from "./getInputStyle";

const StyledInput = styled.input<any>`
  ${getInputStyle};
`;
StyledInput.displayName = "StyledInput";

const InputField = (props: InputFieldProps) => {
  const {
    inputRef,
    appearance = "standard",
    type = "text",
    name,
    className,
    id,
    autoComplete = "off",
    size = "md",
    tabIndex,
    value,
    defaultValue,
    isRequired,
    isError,
    isValid,
    isWarning,
    isFocus = false,
    isDisabled = false,
    onChange,
    onBlur,
    ...rest
  } = props;

  let input!: HTMLInputElement | undefined;

  const getInputRef = (ref: HTMLInputElement) => {
    input = ref;
    if (inputRef) inputRef(ref);
  };

  React.useEffect(() => {
    if (isFocus && input) input.focus();
    return () => (input = undefined);
  });

  return (
    <>
      <StyledInput
        ref={getInputRef}
        type={type}
        size={size}
        name={name}
        appearance={appearance}
        autoComplete={autoComplete}
        className={className}
        id={id}
        required={isRequired}
        value={value}
        defaultValue={defaultValue}
        isFocus={isFocus}
        isValid={isValid}
        isError={isError}
        isWarning={isWarning}
        disabled={isDisabled}
        onChange={isDisabled ? undefined : onChange}
        onBlur={onBlur}
        tabIndex={tabIndex}
        {...rest}
      />
    </>
  );
};

export default InputField;
