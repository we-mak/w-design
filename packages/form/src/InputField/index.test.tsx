import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import { Provider } from "@w-design/core";
import InputField from "./index";

afterEach(cleanup);

describe("<AvatarGroup/>", () => {
  // Render test
  it("should renders correctly", () => {
    const { container } = render(
      <Provider>
        <InputField placeholder="Default input" />
        <InputField size="lg" placeholder="Large input" />
        <InputField size="sm" placeholder="Small input" />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
