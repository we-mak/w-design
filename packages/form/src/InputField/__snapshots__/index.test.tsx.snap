// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarGroup/> should renders correctly 1`] = `
<div>
  <input
    autocomplete="off"
    class="sc-bdfBwQ bbRCOV"
    placeholder="Default input"
    type="text"
    value=""
  />
  <input
    autocomplete="off"
    class="sc-bdfBwQ dMkIpY"
    placeholder="Large input"
    type="text"
    value=""
  />
  <input
    autocomplete="off"
    class="sc-bdfBwQ fqaTNT"
    placeholder="Small input"
    type="text"
    value=""
  />
</div>
`;
